gems/aws-sdk-core/lib/aws-sdk-core/plugins/stub_responses.rb

```diff
@@ -49,17 +49,21 @@ def after_initialize(client)
       class Handler < Seahorse::Client::Handler
 
         def call(context)
-          stub = context.client.next_stub(context)
-          resp = Seahorse::Client::Response.new(context: context)
-          async_mode = context.client.is_a? Seahorse::Client::AsyncBase
-          if Hash === stub && stub[:mutex]
-            stub[:mutex].synchronize { apply_stub(stub, resp, async_mode) }
-          else
-            apply_stub(stub, resp, async_mode)
+          span_wrapper(context) do
+            stub = context.client.next_stub(context)
+            resp = Seahorse::Client::Response.new(context: context)
+            async_mode = context.client.is_a? Seahorse::Client::AsyncBase
+            if Hash === stub && stub[:mutex]
+              stub[:mutex].synchronize { apply_stub(stub, resp, async_mode) }
+            else
+              apply_stub(stub, resp, async_mode)
+            end
+            async_mode ? Seahorse::Client::AsyncResponse.new(
+              context: context,
+              stream: context[:input_event_stream_handler].event_emitter.stream,
+              sync_queue: Queue.new
+            ) : resp
```

Comment:
If we're wrapping this in a block, probably best to refactor this into a method or methods now.